scripts/projectUpdater.ps1

```diff
@@ -102,9 +102,82 @@ if ([string]::IsNullOrEmpty($acceptAll)) {
     }
 }
 
-# copy the new one and make the subs
-$templateName = Get-Content $projectFolder/.conf/.template
-$containerName = Get-Content $projectFolder/.conf/.container
+# PROJECT UPDATER - The first thing is to update it
+if (
+    -not (_checkIfFileContentIsEqual `
+            $Env:HOME/.apollox/scripts/projectUpdater.ps1 `
+            $projectFolder/.conf/projectUpdater.ps1)
+) {
+    # in this case we need to update the project updater
+    # and then run it again
+    Copy-Item `
+        $Env:HOME/.apollox/scripts/projectUpdater.ps1 `
+        $projectFolder/.conf/projectUpdater.ps1
+
+    Write-Host `
+        -ForegroundColor DarkYellow `
+        "⚠️  project updater updated, running it again"
+
+    # run the project updater again
+    & $projectFolder/.conf/projectUpdater.ps1 `
+        $projectFolder `
+        $projectName `
+        $acceptAll `
+        $true
+
+    exit $LASTEXITCODE
+}
+
+# get the metadata of templates.json
+$_templatesJson = Get-Content "$Env:HOME/.apollox/templates.json" | ConvertFrom-Json
+
+# Replace the the .template and .container with metadata.json
+if ((Test-Path "$projectFolder/.conf/.template") -and (Test-Path "$projectFolder/.conf/.container")) {
+    $templateName = Get-Content $projectFolder/.conf/.template
+    $containerName = Get-Content $projectFolder/.conf/.container
+
+    $_metadataJson = New-Object PSObject
+    $_metadataJson | Add-Member -MemberType NoteProperty -Name "templateName" -Value $templateName
+    $_metadataJson | Add-Member -MemberType NoteProperty -Name "containerName" -Value $containerName
+    # If this property doesn't exist, then we are on Torizon 6
+    if ($null -eq $_templatesJson.TorizonOSMajor) {
+        $_torizonOSMajor = "6"
+    } else {
+        $_torizonOSMajor = $_templatesJson.TorizonOSMajor
+    }
+    $_metadataJson | Add-Member -MemberType NoteProperty -Name "torizonOSMajor" -Value $_torizonOSMajor
+
+    # Save the modified JSON object to a file
+    Set-Content -Path "$projectFolder/.conf/metadata.json" -Value ($_metadataJson | ConvertTo-Json) -Encoding UTF8
+
+    Remove-Item -Path $projectFolder/.conf/.template -Force
+    Remove-Item -Path $projectFolder/.conf/.container -Force
+}
+
+# get the metadata of the project
+$_metadataJson = Get-Content "$projectFolder/.conf/metadata.json" | ConvertFrom-Json
+$templateName = $_metadataJson.templateName
+$containerName = $_metadataJson.containerName
+$_torizonOSMajor = $_metadataJson.TorizonOSMajor
+
+# If it's not the current version, it's because the person has torizon.templatesBranch setting set
+if ($_templatesJson.TorizonOSMajor -ne "7") {
+    Write-Host -ForegroundColor DarkYellow "The current Torizon OS version is 7. If you want to upgrade to the current version, remove the torizon.templatesBranch setting."
+}
```

Comment:
can be the tag too.
IMHO this is not needed since we are asking the user that a major update will happen, and also asking for their approval.

Reply:
This is for the cases where the user has forgot that he has the `torizon.templatesBranch`, so he needs to be reminded to remove it.

Reply:
but if is really a major update this will make more confusion.

Reply:
But this is shown just if he is on 6 and updates to 6. It's just a reminder that the current version of Torizon OS is the 7, and if he wants to update to it he needs to remind to remove that setting